src/core/desktopentry.cpp

```diff
@@ -331,44 +309,51 @@ void DesktopEntryManager::scanPath(const QDir& dir, const QString& prefix) {
 				continue;
 			}
 
-			auto id = prefix + entry.fileName().sliced(0, entry.fileName().length() - 8);
-			auto lowerId = id.toLower();
+			auto id = manager->extractIdFromPath(entry.absoluteFilePath());
+			auto content = QString::fromUtf8(file.readAll());
 
-			auto text = QString::fromUtf8(file.readAll());
-			auto* dentry = new DesktopEntry(id, this);
-			dentry->parseEntry(text);
-
-			if (!dentry->isValid()) {
-				qCDebug(logDesktopEntry) << "Skipping desktop entry" << path;
-				delete dentry;
-				continue;
-			}
+			ParsedDesktopEntry parsed;
+			parsed.id = id;
+			parsed.content = std::move(content);
```

Comment:
Desktop entry content should be parsed off-thread. Use bindable properties to update the fields of desktopentry/desktopaction.

Reply:
Doing the parsing in DesktopEntryScanner thread now, and made properties bindable, update them in onScanCompleted.